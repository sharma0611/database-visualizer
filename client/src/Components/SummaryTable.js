// @flow
import React from 'react'

//components
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import { Box, Flex } from 'rebass'
import LoadingSpinner from './LoadingSpinner'
import Error from './Error'
import FlexPaper from './FlexPaper'

// types
import type { SummaryRow, IsLoading, HasErrored } from '../Types/summary'

type Props = {
    readableVariable: ?string,
    rows: ?Array<SummaryRow>,
    isLoading: IsLoading,
    hasErrored: HasErrored
}

const SummaryTable = (props: Props) => {
    const { readableVariable: variable, rows, isLoading, hasErrored } = props

    if (!variable) {
        return null
    }

    if (isLoading) {
        return (
            <Flex justifyContent="center">
                <LoadingSpinner />
            </Flex>
        )
    }

    if (hasErrored) {
        return (
            <Box>
                <Error />
            </Box>
        )
    }

    if (!rows) {
        return null
    }

    return (
        <FlexPaper>
            <Table>
                <TableHead>
                    <TableRow>
                        <TableCell>{variable}</TableCell>
                        <TableCell align="right">Count</TableCell>
                        <TableCell align="right">Average Age</TableCell>
                    </TableRow>
                </TableHead>
                <TableBody>
                    {rows.map(row => (
                        <TableRow key={row.value}>
                            <TableCell component="th" scope="row">
                                {row.value}
                            </TableCell>
                            <TableCell align="right">{row.count}</TableCell>
                            <TableCell align="right">{row.averageAge}</TableCell>
                        </TableRow>
                    ))}
                </TableBody>
            </Table>
        </FlexPaper>
    )
}

export default SummaryTable
