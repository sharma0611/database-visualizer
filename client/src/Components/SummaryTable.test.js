import React from 'react'
import SummaryTable from './SummaryTable'
import renderer from 'react-test-renderer'

it('renders correctly for no variable', () => {
    const props = {
        readableVariable: null,
        isLoading: false,
        hasErrored: false,
        rows: []
    }
    const component = renderer.create(<SummaryTable {...props} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})

it('renders correctly for a variable', () => {
    const props = {
        readableVariable: 'education',
        isLoading: false,
        hasErrored: false,
        rows: [
            { value: 'high school', count: 10, age: 19.3 },
            { value: 'college', count: 70, age: 25.3 }
        ]
    }
    const component = renderer.create(<SummaryTable {...props} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})

it('renders correctly for error state', () => {
    const props = {
        readableVariable: 'education',
        isLoading: false,
        hasErrored: true,
        rows: []
    }
    const component = renderer.create(<SummaryTable {...props} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})

it('renders correctly for loading state', () => {
    const props = {
        readableVariable: 'education',
        isLoading: true,
        hasErrored: false,
        rows: []
    }
    const component = renderer.create(<SummaryTable {...props} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
