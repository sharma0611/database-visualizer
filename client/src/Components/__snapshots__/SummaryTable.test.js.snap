// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for a variable 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
>
  <table
    className="MuiTable-root-28"
  >
    <thead
      className="MuiTableHead-root-29"
    >
      <tr
        className="MuiTableRow-root-30 MuiTableRow-head-33"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root-35 MuiTableCell-head-36"
          scope="col"
        >
          education
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-35 MuiTableCell-head-36 MuiTableCell-alignRight-45"
          scope="col"
        >
          Count
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-35 MuiTableCell-head-36 MuiTableCell-alignRight-45"
          scope="col"
        >
          Average Age
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root-47"
    >
      <tr
        className="MuiTableRow-root-30"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root-35 MuiTableCell-body-37"
          scope="row"
        >
          high school
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root-35 MuiTableCell-body-37 MuiTableCell-alignRight-45"
        >
          10
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-35 MuiTableCell-body-37 MuiTableCell-alignRight-45"
        />
      </tr>
      <tr
        className="MuiTableRow-root-30"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root-35 MuiTableCell-body-37"
          scope="row"
        >
          college
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root-35 MuiTableCell-body-37 MuiTableCell-alignRight-45"
        >
          70
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-35 MuiTableCell-body-37 MuiTableCell-alignRight-45"
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders correctly for error state 1`] = `
<div
  className="sc-bdVaJa iHZvIS"
>
  <h2
    className="sc-bdVaJa sc-htpNat sc-bxivhb jUvXyu"
    color="blue.1"
    fontSize={
      Array [
        6,
      ]
    }
    fontWeight="bold"
  >
    An error has been encountered. Please try again.
  </h2>
</div>
`;

exports[`renders correctly for loading state 1`] = `
<div
  className="sc-bdVaJa iHZvIS"
>
  <div
    className="sc-htoDjs hpxSNI"
  >
    <div
      class="spinner"
    >
      <div
        class="rect rect1"
      />
      <div
        class="rect rect2"
      />
      <div
        class="rect rect3"
      />
      <div
        class="rect rect4"
      />
      <div
        class="rect rect5"
      />
    </div>
  </div>
</div>
`;

exports[`renders correctly for no variable 1`] = `
<div
  className="sc-bdVaJa iHZvIS"
>
  <h2
    className="sc-bdVaJa sc-htpNat sc-bxivhb jUvXyu"
    color="blue.1"
    fontSize={
      Array [
        6,
      ]
    }
    fontWeight="bold"
  >
    No Variable Selected
  </h2>
</div>
`;
